File: 프로그래머스/lv0/120897. 약수 구하기/약수 구하기.java
class Solution {
    public int[] solution(int n) {
       int count = 0;
		
		for(int i = 1; i <= n; i++) {
			if(n % i == 0) {
				count++;
			} // 약수의 갯수를 count해준다			
		}
		int[] answer = new int[count];
		// 약수가 담길 배열의 길이는 count로 정한다
		int j = 0;
		for(int i = 1; i <= n; i++) {
			if(n % i == 0) {
				answer[j] = i;
				j++;
			} // 약수를 answer에 담아준다
		}
        return answer;
    }
}

File: 프로그래머스/0/120897. 약수 구하기/약수 구하기.js
function solution(n) {
    let count = 0;

    for (let i = 1; i <= n; i++) {
        if (n % i === 0) {
            count++;
        } // 약수의 갯수를 count해준다
    }
    let answer = new Array(count);
    // 약수가 담길 배열의 길이는 count로 정한다
    let j = 0;
    for (let i = 1; i <= n; i++) {
        if (n % i === 0) {
            answer[j] = i;
            j++;
        } // 약수를 answer에 담아준다
    }
    return answer;
}
